merkle/src/merkle.rs

```diff
@@ -852,6 +892,14 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
         }
     }
 
+    pub fn read_into_all(&self, pos: usize, buf: &mut [u8]) {
```

Comment:
This would either then read all of the leaves _or_ the top half, but not both, the _all_ might be a bit misleading here.